### DISCUSSION TITLE
Lag Compensation Misinformation on Docs

### DISCUSSION THREAD

**Discussion by Jazz23** [FirstGearGames/FishNet#721] [Documentation]
The [projectile lag compensation example](https://fish-networking.gitbook.io/docs/manual/guides/lag-compensation/projectiles) passes `TimeManager.Tick` into the `ServerFire` RPC to calculate time passed. I recently tried to implement this in 4.3.7 and the `TimeManager.Tick` value is *already* lag compensated, which results in `TimeManager.TimePassed` returning `0`.

The proper way for the client to send it's current tick at the time of fire would be to follow the guide for [rollback raycasting](https://fish-networking.gitbook.io/docs/manual/guides/lag-compensation/raycast), which sends a `PreciseTick` using the last received packet. The projectile lag compensation example should instead look something like `ServerFire(position, direction, base.TimeManager.GetPreciseTick(TickType.LastPacketTick).Tick);` Realistically you would want to send the `PreciseTick` as it contains the `PercentAsDouble` field for a more accurate time discrepancy measurement.

Funny enough, the raycast page in the docs linked above also contains a typo. The syntax `PreciseTick pt = base.TimeManager.GetPreciseTick(base.TimeManager.LastPacketTick);` no longer compiles, and should be replaced with `PreciseTick pt = base.TimeManager.GetPreciseTick(TickType.LastPacketTick);`

**Comment by FirstGearGames** [comment]
Tick is a guess as to what the current tick is on the server. 

If client fires on Tick 100 and it takes 5 ticks to get to the server, the server will then be on ~tick 105. Subtracting 100 from 105 = 5 ticks passed.

I'll check into the API being wrong on the docs though, ty!

**Comment by FirstGearGames** [comment]
There was a bug discovered that Tick was giving the incorrect value. This has been fixed in 4.5.9.
Guides were updated as well at some point. Thank you!
